test/CMakeLists.txt

```diff
@@ -67,6 +70,18 @@ macro(onedpl_add_test test_source_file)
         target_compile_options(${_test_name} PRIVATE /bigobj)
     endif()
 
+    # Disable checked iterators on Windows for debug mode
+    # For details please see
+    #    https://learn.microsoft.com/en-us/cpp/standard-library/iterator-debug-level?view=msvc-170
+    #    https://learn.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170
+    #    https://stackoverflow.com/questions/51494506/replacing-md-with-mt-in-cmake-is-not-possible-in-release-mode
+    if (WIN32 AND ${_build_type_in_lower} STREQUAL "debug")
```

Comment:
This breaks support for calling CMake without a build type specified, such as:
```bash
$ cmake -G "Ninja" -DCMAKE_CXX_COMPILER=icx -DONEDPL_BACKEND=dpcpp ..
```
This behavior is currently support, but with this patch now results in the following error:
```
CMake Error at test/CMakeLists.txt:78 (if):
  if given arguments:

    "WIN32" "AND" "STREQUAL" "debug"

  Unknown arguments specified
```

Reply:
How to reproduce this error?